spec/unit/provider/package/zypper_spec.rb

```diff
@@ -163,6 +165,15 @@ def shell_out_expectation!(*command, **options)
       )
       provider.install_package(["emacs"], ["1.0"])
     end
+
+    it "should run zypper install with source option" do
+      new_resource.source "/tmp/wget_1.11.4-1ubuntu1_amd64.rpm"
+      allow(::File).to receive(:exist?).with("/tmp/wget_1.11.4-1ubuntu1_amd64.rpm").and_return(true)
+      shell_out_expectation!(
+        "zypper", "--non-interactive", "install", "--auto-agree-with-licenses", "--oldpackage", "/tmp/wget_1.11.4-1ubuntu1_amd64.rpm"
+      )
+      provider.install_package(["wget"], ["1.11.4-1ubuntu1_amd64"])
+    end
```

Comment:
Could there be a test here showing what happens when:
1. when the version is given but does not match the source package

Reply:
I vaguely seem to remember us discussing this a few weeks back before Lamont left, and we decided it was undefined behavior. Currently, the `version` property is only used to determine if an action is needed (doesn't it match the installed version), and then if it doesn't, we install whatever the source is. If the source isn't the right version, this of course won't be idempotent and it'll run again on the next run. But AFAICT, we've never checked for this in any provider and the behavior is undefined.

Reply:
@PrajaktaPurohit  Version attributed is overlooked if source is provided in any recipe and if the functionality has to change then it should be raised as a separate issue according to me. This is how its working for all the other package handlers like dpkg, rpm .. I have tested the same.